common/include/villas/kernel/devices/ip_device_reader.hpp

```diff
@@ -0,0 +1,38 @@
+/* Devicetree reader which parses fpga ip devices from a filesystem path
+ *
+ * Author: Pascal Bauer <pascal.bauer@rwth-aachen.de>
+ *
+ * SPDX-FileCopyrightText: 2023-2024 Pascal Bauer <pascal.bauer@rwth-aachen.de>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <filesystem>
+#include <stdexcept>
+#include <vector>
+
+#include <villas/kernel/devices/ip_device.hpp>
+#include <villas/utils.hpp>
+
+class IpDeviceReader {
```

Comment:
Why do you create a class for this? Wouldn't this make more sense as a function?